src/main/kotlin/uk/gov/justice/digital/hmpps/manageoffencesapi/service/AdminService.kt

```diff
@@ -75,6 +83,61 @@ class AdminService(
     }
   }
 
+  @Transactional
+  fun createEncouragementOffence(parentOffenceId: Long): Offence {
+    val offence = offenceRepository.findById(parentOffenceId)
+      .orElseThrow { EntityNotFoundException("Offence not found with ID $parentOffenceId") }
+
+    // Parent offence should have no parentCode, or parentCode must be the same as the current Offence (indicating the parent)
+    if (offence.parentCode !== null && offence.parentCode !== offence.code) {
+      throw ValidationException("Offence must be a valid parent")
+    }
+
+    if (offence.endDate !== null && offence.endDate < encouragementOffenceEligibilityStartDate) {
+      throw ValidationException("Offence must have an end date post $encouragementOffenceEligibilityStartDate")
+    }
+
+    val children = offenceRepository.findByParentOffenceId(parentOffenceId)
+    val encouragementCode = offence.code + 'E'
+
+    if (children.any { it.code == encouragementCode }) {
+      throw ValidationException("Encouragement offence already exists for offence code $encouragementCode")
+    }
+
+    val prisonRecord = prisonApiClient.findByOffenceCodeStartsWith(offence.code, 0)
+
+    if (prisonRecord.content.isEmpty()) {
+      throw ValidationException("No prison record was found for offence code ${offence.code}")
+    }
+
+    val encouragementOffence = offence.copy(
```

Comment:
Can you also explicitly set the createdDate, lastUpdateDate and changedDate

Reply:
Good catch! I'll update the dates.